parser/src/powdr.lalrpop

```diff
@@ -681,6 +683,25 @@ PatternList: Vec<Pattern> = {
 
 // ---------------------------- Trait/Impl Declaration -----------------------------
 
+GenericTraitName: (String, Option<TraitScheme<Expression>>) = {
+    <name:Identifier> => (name, None),
+    <vars:("<" <TypeVarBounds> ">")?> <name:Identifier> <types:("<" <TypeTermList> ">")> =>
+        (name, Some(TraitScheme{ vars: vars.unwrap_or_default(), types }))
+}
+
+ImplDeclaration: TraitImplementation<Expression> = {
+    "impl" <type_scheme: GenericTraitName> "{" <functions:ImplFunctions> "}" => TraitImplementation { name: type_scheme.0, type_scheme: type_scheme.1, functions }
+}
+
+ImplFunctions: Vec<NamedExpression<Expression>> = {
```

Comment:
We could use the same grammatic element (`{`/`}`-separated list of fields with values) for struct expressions